Postgres url does not load

In my play app I have this config

```hocon
slick.dbs.default = {
  driver = "slick.driver.PostgresDriver$"
  db = {
    dataSourceClass = slick.jdbc.DatabaseUrlDataSource
    properties = {
      driver = org.postgresql.Driver
      url = "postgres://username:password@host:port/dbname"
    }
  }
}
```

However, I am getting an error where it wants `slick.dbs.default.db.url`:
`com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'db.url'`

I was told that with Play-Slick 1.1.0, I can use postgres url instead of jdbc url, and settung `dataSourceClass` will skip `slick.dbs.default.db.url`, but obviously it is not the case.

Here it is
```play.api.UnexpectedException: Unexpected exception[PSQLException: ERROR: relation "play_evolutions" does not exist
  Position: 64]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:165) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:121) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.Option.map(Option.scala:146) ~[scala-library-2.11.7.jar:na]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:121) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:119) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.util.Success.flatMap(Try.scala:231) ~[scala-library-2.11.7.jar:na]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:119) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:111) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.11.7.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.11.7.jar:na]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[na:1.8.0_60]
Caused by: org.postgresql.util.PSQLException: ERROR: relation "play_evolutions" does not exist
  Position: 64
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2182) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1911) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:173) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:616) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:452) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:332) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at com.zaxxer.hikari.proxy.StatementProxy.executeQuery(StatementProxy.java:99) ~[HikariCP-java6-2.3.7.jar:na]
	at com.zaxxer.hikari.proxy.StatementJavassistProxy.executeQuery(StatementJavassistProxy.java) ~[HikariCP-java6-2.3.7.jar:na]
	at play.api.db.evolutions.DatabaseEvolutions.executeQuery(EvolutionsApi.scala:298) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.databaseEvolutions(EvolutionsApi.scala:125) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:97) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:112) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DefaultEvolutionsApi.scripts(EvolutionsApi.scala:77) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$play$api$db$evolutions$ApplicationEvolutions$$runEvolutions$1.apply$mcV$sp(ApplicationEvolutions.scala:50) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.withLock(ApplicationEvolutions.scala:98) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.play$api$db$evolutions$ApplicationEvolutions$$runEvolutions(ApplicationEvolutions.scala:49) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at scala.collection.immutable.Stream.foreach(Stream.scala:594) ~[scala-library-2.11.7.jar:na]
	at play.api.db.evolutions.ApplicationEvolutions.start(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.<init>(ApplicationEvolutions.scala:149) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get$lzycompute(EvolutionsModule.scala:53) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:53) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:44) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81) ~[guice-4.0.jar:na]
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61) ~[guice-4.0.jar:na]
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-4.0.jar:na]
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:205) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:199) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:199) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:180) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:96) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:73) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:62) ~[guice-4.0.jar:na]
	at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:126) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:93) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:153) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:150) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:150) ~[play-server_2.11-2.4.3.jar:2.4.3]
	... 14 common frames omitted
```

The bug seems to be related to the creation of `play_evolutions` table. Perhaps when it try to create that table, it has the wrong config. Once I have my `play_evolutions` table created, any subsequent update to the schema is picked up without problem with my config as above.

--- a/src/core/src/test/scala/play/api/db/slick/SlickApiSpec.scala
+++ b/src/core/src/test/scala/play/api/db/slick/SlickApiSpec.scala
@@ -48,7 +48,7 @@ class SlickApiSpec extends Specification {
   "SlickApi.dbConfigs" should {
     "return all DatabaseConfig instances for a valid configuration" in {
       import SUTWithGoodConfig._
-      api.dbConfigs[BasicProfile] must have size(3)
+      api.dbConfigs[BasicProfile] must have size(4)
     }
     "throw if a database name doesn't exist in the config" in {
       import SUTWithBadConfig._

--- a/src/core/src/test/scala/play/api/db/slick/TestData.scala
+++ b/src/core/src/test/scala/play/api/db/slick/TestData.scala
@@ -3,6 +3,15 @@ package play.api.db.slick
 import play.api.Configuration
 
 object TestData {
+  private lazy val h2DatasourceConfig: Configuration = Configuration.from(
+    Map(
+      "slick.dbs.h2datasource.driver" -> "slick.driver.H2Driver$",
+      "slick.dbs.h2datasource.db.dataSourceClass" -> "slick.jdbc.DatabaseUrlDataSource",
+      "slick.dbs.h2datasource.db.properties.driver" -> "org.h2.Driver",
+      "slick.dbs.h2datasource.db.properties.url" -> "jdbc:h2:mem:"
+    )
+  )
+
   val configuration: Configuration = Configuration.from(
     Map(
       "slick.dbs.somedb.driver" -> "slick.driver.H2Driver$",
@@ -14,7 +23,7 @@ object TestData {
 
       "slick.dbs.jdbc-driver-not-recognized.driver" -> "slick.driver.MySQLDriver$",
       "slick.dbs.jdbc-driver-not-recognized.db.driver" -> "play.api.db.slick.SomeDummyDriver"
-  ))
+  )) ++ h2DatasourceConfig
 
   val badConfiguration: Configuration = configuration ++ Configuration.from(
     Map("slick.dbs.missing-slick-driver.db.driver" -> "play.api.db.slick.SomeDummyDriver"))

--- a/src/evolutions/src/main/scala/play/api/db/slick/evolutions/internal/DBApiAdapter.scala
+++ b/src/evolutions/src/main/scala/play/api/db/slick/evolutions/internal/DBApiAdapter.scala
@@ -1,6 +1,7 @@
 package play.api.db.slick.evolutions.internal
 
 import java.sql.Connection
+import java.sql.SQLException
 
 import javax.inject.Inject
 import javax.sql.DataSource
@@ -51,7 +52,7 @@ private[evolutions] object DBApiAdapter {
       }
     }
 
-    lazy val url: String = dbConfig.config.getString("db.url")
+    lazy val url: String = withConnection(_.getMetaData().getURL())
 
     def getConnection(): Connection = dbConfig.db.source.createConnection()
 
@@ -65,7 +66,7 @@ private[evolutions] object DBApiAdapter {
       val conn = getConnection()
       try block(conn)
       finally {
-        try conn.close() catch { case _: java.sql.SQLException => }
+        try conn.close() catch { case _: SQLException => }
       }
     }
 

--- a/src/evolutions/src/test/scala/play/api/db/slick/evolutions/DBApiAdapterSpec.scala
+++ b/src/evolutions/src/test/scala/play/api/db/slick/evolutions/DBApiAdapterSpec.scala
@@ -54,8 +54,12 @@ class DBApiAdapterSpec extends Specification {
     }
     
     "url" should {
-      "return the value set in the config" in {
+      "return the value set in the config for the jdbc url" in {
         db.url must_== TestData.configuration.getString(s"slick.dbs.${dbName}.db.url").get
+      }
+      "return the value set in the config for the datasource url" in {
+        val h2DatasourceDb = api.database("h2datasource")
+        h2DatasourceDb.url must_== TestData.configuration.getString(s"slick.dbs.h2datasource.db.properties.url").get
       }
     }
   }

